### PR TITLE
Make an exception for BTC/sBTC/WBTC

### LINKED ISSUE #2236: Optimism: sBTC balance not displayed after an in-wallet swap
### Discord Discussion Link

https://discord.com/channels/667044843901681675/1019997761497743430/1020665795853946990

### What browsers are you seeing the problem on?

Chrome

### What were you trying to do?

Swap to sBTC, in-wallet swap

Nax: I've reproduced this users issue. Below details are not from the user

### What did not work?

sBTC balance is not diplayed.

tx: https://optimistic.etherscan.io/tx/0x85ba75e11a78f45759ae2c6ca4aad732dcd9767b854841eb3c6261b0742963b3
Screen Shot 2022-09-18 at 13 36 53

Screen Shot 2022-09-18 at 13 37 14

Screen Shot 2022-09-18 at 13 37 35

[logs_v0_16_0__2022-09-18T13_39_11+02_00.txt](https://github.com/tallycash/extension/files/9593865/logs_v0_16_0__2022-09-18T13_39_11%2B02_00.txt)

### Version

v0.16.0

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by kkosiorowska**
Closes #2236 

This PR makes an exception for BTC/sBTC/WBTC and allows it to display more decimal places.

For some assets `unitPrice` is `undefined`.  For such cases, `heuristicDesiredDecimalsForUnitPrice` returns the value declared in `desiredDecimals`  (currently value 2). To be able to display the values of such assets as BTC/sBTC/WBTC the `desiredDecimals` should be greater when the `unitPrice` is `undefined`.

Screenshot 2022-11-04 at 16 03 01

To Test:
- [ ] check that the sBTC balance is visible on the wallet assets list

Latest build: [extension-builds-2556](https://github.com/tallycash/extension/suites/9166504859/artifacts/426869988) (as of Mon, 07 Nov 2022 13:56:49 GMT).

### CODE REVIEW

**Comment by Shadowfiend** [comment]
This is what `heuristicDesiredDecimalsForUnitPrice` does. I believe in these cases we're missing a unit price maybe?

It would be useful to have the reason reflected in the PR description 🙏

**Comment by kkosiorowska** [comment]
> This is what `heuristicDesiredDecimalsForUnitPrice` does. I believe in these cases we're missing a unit price maybe?
> 
> It would be useful to have the reason reflected in the PR description 🙏

Yes in these cases we are missing the unit price. I have added an explanation in the description.

File: background/redux-slices/selectors/accountsSelectors.ts [hunk 1]
```diff
@@ -48,7 +48,14 @@ import {
...
 // TODO What actual precision do we want here? Probably more than 2
 // TODO decimals? Maybe it's configurable?
-const desiredDecimals = 2
+const desiredDecimals = {
+  default: 2,
+  greater: 6,
+}
+
+// List of assets by symbol that should be displayed with more decimal places
+const EXCEPTION_ASSETS_BY_SYMBOL = ["BTC", "sBTC", "WBTC"]
```
Comment (jagodarybacka) [inline background/redux-slices/selectors/accountsSelectors.ts:57]: Do we need to make it case-insensitive?

Comment (kkosiorowska) [reply background/redux-slices/selectors/accountsSelectors.ts]: Good point, let's change it.

**Review by greg-nagy — APPROVED** [review]
works like a charm!

**Pull request merged**